dashboard/app/helpers/lesson_import_helper.rb

```diff
@@ -0,0 +1,213 @@
+    sorted_matches = matches.sort_by {|m| m[:index]}
+    return [ActivitySection.new(description: activity_markdown.strip, seeding_key: SecureRandom.uuid, position: 1)] if matches.empty?
+    sorted_matches = find_markdown_chunks(activity_markdown, sorted_matches)
+    sections = []
+    tip_match_map = {}
+    slide = false
+    tip_matches.each do |match|
+      key = match[:match][3] || "#{match[:match][1]}-0"
+      tip_match_map[key] = match
+    end
+    sorted_matches.each_with_index do |match, i|
+      activity_section = nil
+      if match[:type] == 'skippable' || match[:type] == 'tip'
+        next
+      elsif match[:type] == 'slide'
+        # For now, if we find the slide token, apply it to the next thing
+        slide = true
+        next
+      elsif match[:type] == 'tiplink'
+        activity_section = create_activity_section_with_tip(match[:match], tip_match_map)
+      elsif match[:type] == 'remark'
+        activity_section = create_activity_section_with_remark(match[:match])
+      else
+        activity_section = ActivitySection.new(description: match[:substring].strip)
+      end
+      next unless activity_section
+      activity_section.position = i + 1
+      activity_section.slide = slide
+      slide = false
+      activity_section.seeding_key ||= SecureRandom.uuid
+      sections = sections.push(activity_section)
+    end
+    tip_match_map.each do |key, value|
+      match = value[:match]
+      activity_section = ActivitySection.new
+      activity_section.position = sections.length + 1
+      activity_section.seeding_key ||= SecureRandom.uuid
+      activity_section.tips = [create_tip(key, match[1] || "tip", match[4] || "no markdown found")]
+      sections.push(activity_section)
+    end
+    sections
+  end
+
+  def self.create_lesson_activities(activities_data, levels, lesson_id)
+    activities = activities_data.map.with_index(1) do |a, i|
+      @lesson_activity = LessonActivity.new
+      @lesson_activity.name = a['name']
+      @lesson_activity.duration = a['duration'].split[0].to_i
+      @lesson_activity.lesson_id = lesson_id
+      @lesson_activity.seeding_key = SecureRandom.uuid
+      @lesson_activity.position = i
+      @lesson_activity.save!
+      @lesson_activity.reload
+      @lesson_activity.activity_sections = create_activity_sections(a['content'])
+      @lesson_activity
+    end
+
+    # Create a lesson with all the levels in them
+    # TODO use the [code-studio] syntax from CB instead
+    @lesson_activity = LessonActivity.new
+    @lesson_activity.name = "Levels"
+    @lesson_activity.lesson_id = lesson_id
+    @lesson_activity.seeding_key = SecureRandom.uuid
+    @lesson_activity.position = activities.length + 1
+    @lesson_activity.save!
+    @lesson_activity.reload
+    @lesson_activity.activity_sections = [create_activity_section_with_level_ranges(0, levels.count - 1, levels)]
+    activities.push(@lesson_activity)
```

Comment:
I'm a bit concerned about `@lesson_activity` as a class variable. it looks you should be able to:
1. declare it as `lesson_activity = nil` at the beginning of the method
2. pass it as a parameter to create_activity_section_with_level_ranges

Reply:
Yeah, I got bitten by this bad practice already haha. I'll move away from it.